deber/TeoremaMaestro.tex

```diff
@@ -0,0 +1,164 @@
+\documentclass[12pt, letterpaper]{article} %Declarando clase de documento, tamaño
+%=====Librerias=====%
+\usepackage[utf8]{inputenc}
+
+\usepackage[superscript,biblabel]{cite}
+\usepackage{hyperref}
+\usepackage{graphicx}
+\usepackage{amsmath, amssymb, amsfonts}
+\usepackage{amsthm}
+\usepackage{mathrsfs}
+
+
+%===Definiendo Idioma español
+\usepackage[spanish]{babel}
+\selectlanguage{spanish}
+
+%==============Control de margenes==========
+\usepackage[a4paper,tmargin=3cm,bmargin=2.5cm,lmargin=3cm, rmargin=2.5cm, bindingoffset=6mm]{geometry}
+
+%======Datos Generales======
+\title{Teorema Maestro}
+\author{Johan Andrés Toloza}
+\date{Abril, 2020}
+%=====Documento======%
+\begin{document}
+
+\maketitle
+\section*{Teorema Maestro} \hline
+En el análisis de algoritmos, el \textbf{teorema maestro} proporciona una solución sencilla en términos asintóticos (usando una Cota superior asintótica) para ecuaciones de recurrencias que ocurren en muchos algoritmos recursivos tales como en el Algoritmo divide y vencerás. Fue popularizado por el libro \textit{introducción a los algoritmos} por Cormen, Leiserson, Rivest y Stein, en el cual fue tanto introducido como probado formalmente. No todas las ecuaciones de recurrencia pueden ser resueltas con el uso del teorema maestro.
+
+\tableofcontents
+
+\section{Introducción}
+Considere un problema que puede ser resuelto a través de un algoritmo recursivo como el siguiente:\\
+
+\vspace{1mm}
+\textbf{procedimiento} T(n:tamaño del problema) \textbf{definición:}
```

Comment:
Considera utilizar un paquete para algoritmos
\usepackage[ruled,vlined]{algorithm2e}
\begin{algorithm}[htbp]
\SetAlgoLined
\SetKw{exit}{exit}
\SetKwProg{Fn}{Procedimiento}{:}{end}
\Fn{T(n: tamañoProblema)}{
        \If(){
            $n<1$}{exit\;}
        \Else(){
            \Repeat{$a$ veces}{$T\left(n/b\right)$\;}
        }

    }

 \caption{How to write algorithms}
\end{algorithm}